publish.sh publishes packages in wrong order and swallows failures

scripts/publish.sh loops over $SCRIPT_DIR/../packages/* in filesystem (alphabetical) order instead of the dependency order defined in PACKAGES_TO_PUBLISH. This publishes fingerprint-generator and fingerprint-injector before generative-bayesian-network and header-generator, so pnpm fails with ERR_PNPM_CANNOT_RESOLVE_WORKSPACE_PROTOCOL for their workspace deps.

See run: https://github.com/apify/fingerprint-suite/actions/runs/30079078585/job/89437427113#step:6:663

The script has no set -e and doesn't check pnpm publish's exit code, so these failures are printed but the step still exits 0 and the whole workflow reports success.

Fix: iterate over PACKAGES_TO_PUBLISH directly (in the declared order) instead of globbing packages/*, and add set -e or check each pnpm publish exit code so a failed publish actually fails the job.

--- a/scripts/publish.sh
+++ b/scripts/publish.sh
@@ -1,17 +1,16 @@
 #!/bin/bash
+set -e
 
 SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
 PACKAGES_TO_PUBLISH=( "generative-bayesian-network" "header-generator" "fingerprint-generator" "fingerprint-injector" )
 
-for FOLDER in $SCRIPT_DIR/../packages/*; do
+for PACKAGE_NAME in "${PACKAGES_TO_PUBLISH[@]}"; do
+  FOLDER="$SCRIPT_DIR/../packages/$PACKAGE_NAME"
   if [[ -d $FOLDER ]]; then
-    PACKAGE_NAME=$(basename $FOLDER)
-    if [[ " ${PACKAGES_TO_PUBLISH[@]} " =~ " ${PACKAGE_NAME} " ]]; then
-      echo "Publishing $PACKAGE_NAME"
-      cd $FOLDER/dist
-      pnpm publish --no-git-checks
+    echo "Publishing $PACKAGE_NAME"
+    cd $FOLDER/dist
+    pnpm publish --no-git-checks
 
-      cd $SCRIPT_DIR/../
-    fi
+    cd $SCRIPT_DIR/../
   fi
 done

